CHANGELOG.md

```diff
@@ -6,6 +6,38 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 (modification: no type change headlines) and this project adheres to
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [4.0.0] - 2020-04-17
+
+This release introduces a major API upgrade from callbacks to Promises. Behind the scenes the library and its tests have been refactored to TypeScript.
```

Comment:
Can you add just a super-simple import/usage 3-4 liner example here (eventually just more or less copy the first one from the README) so that one get a better first glimpse on the new API structure?

Reply:
Or - as a more general note - can you be more descriptive what has changed for the user with this release?

Reply:
thanks for the feedback, just pushed an update to the changelog, if you want to see if it is better